runtime/runtime/src/actions.rs

```diff
@@ -438,6 +442,18 @@ pub(crate) fn action_deploy_contract(
         })?;
     account.code_hash = code.get_hash();
     set_code(state_update, account_id.clone(), &code);
+    if let Some(cache) = cache {
+        // We only precompile contract for Wasmer VM at the moment.
+        if let Some(err) =
+            precompile(code.get_code(), &code.get_hash(), vm_config, cache.deref(), VMKind::Wasmer)
+        {
+            return Err(StorageError::StorageInconsistentState(format!(
```

Comment:
A more idiomatic (but not necessary more readable :) ) version would be `

```
precompile(...).map_err(|err| StorageError(...))?
```

Reply:
IIRC originally it was `map_err`, but IDE inspection suggested me to rewrite with 'let`.

Reply:
Ah, sorry, I was confused by the peculiar signature of `precompile`. It returns an `Option<VMError>`. I would expect it to return `Result<(), VMError>`. The tangible benefit there would be that it would be impossible to forget to check the error: compiler warns about unused results, but not about unused options. And map_err would also work then.